Form/DataTransformer/ProviderDataTransformer.php

```diff
@@ -32,9 +38,10 @@ class ProviderDataTransformer implements DataTransformerInterface
      * @param string $class
      * @param array  $options
      */
-    public function __construct(Pool $pool, $class, array $options = array())
+    public function __construct(Pool $pool, $class, LoggerInterface $logger, array $options = array())
```

Comment:
If you want to stay BC, you have to put it last, and if you want to put it last, you have to make it optional. Instead of using `NullLogger`, you can choose to rethrow the exception if the logger is null though.